### ISSUE TITLE
Version 4.x not updating with apt update Ubuntu 20.04

### ISSUE THREAD

**Issue by TCROC** [containers/podman#14302]
On Ubuntu 20.04, Podman is stuck at version 3.4.2.  Running apt update doesn't update it to the 4.x version.  I followed the instructions here: https://podman.io/getting-started/installation.html#linuxmint-20x.

**Comment by afbjorklund** [comment]
Those packages (from Kubic) are not updated anymore, the instructions should be removed (for Mint, like they were for Ubuntu)

**Comment by mythi** [comment]
> Those packages (from Kubic) are not updated anymore, the instructions should be removed (for Mint, like they were for Ubuntu)

has there been any announcements about kubic? we've installed `buildah` from it and that disappeared completely.

**Comment by TCROC** [comment]
> Those packages (from Kubic) are not updated anymore, the instructions should be removed (for Mint, like they were for Ubuntu)

What is the new way to install Podman on Ubuntu 20.04?

**Comment by afbjorklund** [comment]
https://podman.io/blogs/2022/04/05/ubuntu-2204-lts-kubic.html

**Comment by TCROC** [comment]
Ok I see.  So Podman 4.x is available on Ubuntu 22.04 if I upgrade?

**Comment by afbjorklund** [comment]
It is available on Fedora and RHEL, not on Ubuntu (yet)

**Comment by TCROC** [comment]
Any plans for when it will land on Ubuntu? 👀 :)

**Comment by afbjorklund** [comment]
https://packages.debian.org/search?keywords=podman&searchon=names&exact=1&suite=all&section=all

Not in "Sid" yet

**Comment by TCROC** [comment]
Ah ok cool.  And am I able to add that as a PPA on Ubuntu?

**Comment by afbjorklund** [comment]
I believe you would need to build it from the source code.
* https://github.com/containers/podman/issues/13407#issuecomment-1058137350

**Comment by TCROC** [comment]
What is the reasoning for not including 4.x in an Ubuntu ppa?

**Comment by lsm5** [comment]
@TCROC Ubuntu ppas are a huge PITA (at least for me (Fedora guy)).  Also, no debian support from what I can tell.

RE: debian experimental repo that @afbjorklund mentioned above, you should be able to install the deb package on ubuntu, at least that's what the debian maintainer told us. I don't know if you'd be able to add it as a repo source to Ubuntu though.

Also, I have a WIP kubic repo at https://build.opensuse.org/project/show/devel:kubic:libcontainers:unstable . This will likely not be officially documented for quite some time as I *cannot* promise long term maintainability / support yet. But if you're willing to try it out, be my guest. This repo currently allows you to use podman v4.1.0 on debian sid and ubuntu 22.04 LTS along with the latest network stack (netavark and aardvark-dns).

Closing..

**Comment by wuestkamp** [comment]
Is there still a way to install Podman (any version) on Ubuntu 20.04?
Because this throws `gpg: no valid OpenPGP data found`:

```
. /etc/os-release
echo "deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/testing/xUbuntu_${VERSION_ID}/ /" | sudo tee /etc/apt/sources.list.d/devel:kubic:libcontainers:testing.list
curl -L "https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/testing/xUbuntu_${VERSION_ID}/Release.key" | sudo apt-key add -
sudo apt-get update -qq
sudo apt-get -qq -y install podman
```

**Comment by TCROC** [comment]
@wuestkamp I suggested something over here: https://github.com/containers/podman/issues/14332

 Sounds like Snap or Flatpak might be a possibility 👀.  Still waiting on an official comment.

**Comment by afbjorklund** [comment]
> Is there still a way to install Podman (any version) on Ubuntu 20.04?

The kubic packages are still the "best", but they do have security issues.

**Comment by lsm5** [comment]
Given all the love for podman and ubuntu 20.04, I have restored podman on the `devel:kubic:libcontainers:stable` project. Again, this is the same v3.4.2 build that used to exist prior and is vulnerable to the CVEs mentioned above. So, use at your own risk.

Builds should be published in a few mins.

**Comment by crfrolik** [comment]
I'm confused by this. Why is it still so difficult to get 4.X installed on Ubuntu?

**Comment by vrothberg** [comment]
> I'm confused by this. Why is it still so difficult to get 4.X installed on Ubuntu?

It's an immense amount of work.  It is not "only" packaging podman but many of its runtime dependencies including config files, man pages etc.  Plus maintaining all of that.

Since podman is now in the main repositories of Ubuntu, we prefer users to use these packages.

**Comment by alexanderadam** [comment]
> Since podman is now in the main repositories of Ubuntu, we prefer users to use these packages.

The current Ubuntu 22.04 LTS will be supported for 10 years.
So you suggest that its users are stuck with Podman 3 for ten years?

I also [asked on StackOverflow for a solution](https://askubuntu.com/questions/1414446/whats-the-recommended-way-of-installing-podman-4-in-ubuntu-22-04) but so far no one got an answer how to get Podman 4 in a sane manner that doesn't involve setting up and maintaining binaries manually.

It's probably pointless to mention again that an official PPA would solve that for everybody. :wink: 

…but I also [had some issues with Kernel built-in overlayfs](https://twitter.com/alexanderadam__/status/1535536381862330368) so there's that. :shrug:

**Comment by afbjorklund** [comment]
The LTS versions (just like EL) are _supposed_ to ship with old versions, that is sort of their selling point ?

So normally you would have to use something like "backports" (EPEL), to get the latest and greatest...

**Comment by vrothberg** [comment]
> The LTS versions (just like EL) are supposed to ship with old versions, that is sort of their selling point ?

I think that depends on the individual needs.  This is also why there are streams in RHEL for shipping different versions of the container tools.

AFAIK, Ubuntu updates the kernel for LTS versions at some point as well.  The balance of "stability" and "new features" is delicate and maintenance for such a long time is a herculean task.

**Comment by afbjorklund** [comment]
I guess that is now up to the Ubuntu maintainers, how they want to handle the "podman" package... Stay at 3.4 or go to 4.x

Still somewhat ironic that it is easier to upgrade Podman on Mac and on Win, but I guess you can use machine on Linux too.

**Comment by lsm5** [comment]
> It's probably pointless to mention again that an official PPA would solve that for everybody. wink

@alexanderadam I have an *UNSTALBLE* Kubic repo at https://build.opensuse.org/project/show/devel:kubic:libcontainers:unstable .  For the near future, it won't get any more official than that.

This repo gets the latest upstream podman and other packages soon after upstream release. I did try it myself sometime ago and AFAICT, 22.04 should mostly just work.

Not recommended for production use, so please be careful where you use it. Let me know what your experience is like.

**Comment by Korijn** [comment]
@lsm5 Let me just get out of the way that I completely understand your motivations. I just want to make you are aware that this is also blocking public CI environments from adoption of podman v4. Github actions, azure pipelines, etc are all also stuck.

**Comment by lsm5** [comment]
> @lsm5 Let me just get out of the way that I completely understand your motivations and reasons not to provide a stable PPA. It's a big piece of work and time is a limited resource.
> 
> I just want to make sure you are aware that this is also blocking public CI environments from adoption of podman v4. Github actions, azure pipelines, etc are all also stuck.

Yes, see: https://github.com/actions/runner-images/issues/5578 . I'm hoping to make this new kubic repo usable by github actions, but distractions galore.

If anyone would like to help speed this up, do let me know.

**Comment by Korijn** [comment]
> If anyone would like to help speed this up, do let me know.

How can we help?

**Comment by lsm5** [comment]
> How can we help?

@Korijn if you're mainly looking at 22.04, please try using the unstable kubic repo mentioned above and report issues to me. I mainly need end user testing for 22.04 as there are no build / installation / basic usage issues to my knowledge, apart from an annoying `Maintainer` field warning.

For 20.04 and others, if you look at the referenced github actions issue, there are some glibc related issues. I haven't had the chance to take a good look at those, but if you could dumb those down for me and / or send me patches which can be included in the packaging source, that'd be awesome. I can give you further details later about how the fedora sources build ubuntu packages if you're interested.

**Comment by Korijn** [comment]
I can definitely try.

> ... the packaging source, ...

Just checking, is this the right URL to get the packaging sources? https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_20.04/

**Comment by lsm5** [comment]
> Just checking, is this the right URL to get the packaging sources? https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/unstable/xUbuntu_20.04/

@Korijn That link contains the release artifacts AFAICT.

For packaging source, you can begin with: https://build.opensuse.org/project/show/devel:kubic:libcontainers:unstable which has the list of packages and build results.

Navigate to one of the packages, for example podman, it will have a `_service` file with link to the git repo and the branch from where the sources are pulled for building.
https://build.opensuse.org/package/view_file/devel:kubic:libcontainers:unstable/podman/_service?expand=1

```
<services>
  <service name="obs_scm">
    <param name="url">https://gitlab.com/rhcontainerbot/rpms-openqa/podman.git</param>
    <param name="scm">git</param>
    <param name="revision">rawhide</param>
    <param name="extract">podman.spec</param>
  </service>
  <service name="download_files">
    <param name="enforceupstream">yes</param>
  </service>
</services>
```

The files listed with ` <param name="extract">` are extracted from that git repo and get used by OBS.

That combined with the `debbuild` project repository as a dependency added to: https://build.opensuse.org/projects/devel:kubic:libcontainers:unstable/prjconf  help to build the deb packages.


Also, let's create a separate issue (either this github or on https://gitlab.com/rhcontainerbot/rpms-openqa/podman.git to continue this conversation and reference that issue here.

**Comment by afbjorklund** [comment]
Is anyone building 3.4.x ?

**Comment by lsm5** [comment]
> Is anyone building 3.4.x ? (20.04)
> 
> Could use that for cri-o in minikube

does that need to depend on 3.4.x ? /cc @haircommander

**Comment by afbjorklund** [comment]
No, just has less requirements (and uses cni)

**Comment by lsm5** [comment]
> No, just has less requirements (and uses cni). Just thought it would be a good idea to get off 3.4.2, without the 4.x breakage...

@afbjorklund RE: cni, is the crio / minikube requirement something that netavark cannot address?

**Comment by afbjorklund** [comment]
As a Kubernetes node, it will probably use CNI for the foreseeable future

Double network stacks is a possibility but not really a wish - maybe host net ?

**Comment by lsm5** [comment]
@afbjorklund btw, podman 4.x should still be usable with cni. We can probably adjust dependencies to fetch/ depend on cni for that case.

**Comment by afbjorklund** [comment]
I might take a look at it, after upgrading to Ubuntu 22.04. Then again, it will probably be Podman 5.0 by then... Considering to go over to the Docker (compat) socket, to get some backwards compatibility.

**Comment by StarpTech** [comment]
Hello, what's the status here? Is there really no default way to install the latest version of podman on the most popular Linux distros? I know that PPA is a pain, but a shell script would do it as well. Or what about https://github.com/NixOS/nix or brew that is available to Linux and Mac.

**Comment by lsm5** [comment]
> Hello, what's the status here? Is there really no default way to install the latest version of podman on the most popular Linux distros? I know that PPA is a pain, but a shell script would do it as well. Or what about https://github.com/NixOS/nix or brew that is available to Linux and Mac.

status is still https://github.com/containers/podman/issues/14302#issuecomment-1192617223

**Comment by lsm5** [comment]
@Korijn looks like github actions is happy with the version in ubuntu's default repos, so it's unlikely the kubic repo with the latest version will make it there https://github.com/actions/runner-images/issues/5578#issuecomment-1223843419 .  Not sure where the azure pipelines people do things but I won't be surprised if they share that opinion too. 

Nonetheless, I will update the podman installation docs with the unstable Kubic repo for non-production use soon.

**Comment by StarpTech** [comment]
> status is still https://github.com/containers/podman/issues/14302#issuecomment-1192617223

This looks like an unofficial fork. I'd like to see something that is integrated with the guys who maintain this software. I need podman for production use cases.

**Comment by Korijn** [comment]
> @Korijn looks like github actions is happy with the version in ubuntu's default repos, so it's unlikely the kubic repo with the latest version will make it there [actions/runner-images#5578 (comment)](https://github.com/actions/runner-images/issues/5578#issuecomment-1223843419) . Not sure where the azure pipelines people do things but I won't be surprised if they share that opinion too.

They're the same people. Microsoft acquired github a while ago, you may have heard. :) Github actions and azure pipelines use the exact same images.

> Nonetheless, I will update the podman installation docs with the unstable Kubic repo for non-production use soon.

Awesome!

**Comment by lsm5** [comment]
> > status is still [#14302 (comment)](https://github.com/containers/podman/issues/14302#issuecomment-1192617223)
> 
> This looks like an unofficial fork. I'd like to see something that is integrated with the official release flow. I need podman for production use cases.

@StarpTech Sure, unofficial as far as the distro is concerned, but as official as it's gonna get from Podman. Maintained by a podman team member (a.k.a me) on company time. That repo should mostly get the same builds that Fedora does at the same speed as Fedora.

The official debian and ubuntu packages are maintained by volunteers in their spare time, and they could use additional hands and eyes for faster update delivery. The podman team doesn't control those, but I can get you in touch with the right people if you'd be willing to help out.

**Comment by lsm5** [comment]
@StarpTech another option could be download the latest builds from official debian experimental repos. Those often have the latest `official` distro packages, and the binaries being static *should* work on other deb / ubuntu versions.

**Comment by vrothberg** [comment]
> > status is still [#14302 (comment)](https://github.com/containers/podman/issues/14302#issuecomment-1192617223)
> 
> This looks like an unofficial fork. I'd like to see something that is integrated with the official release flow. I need podman for production use cases.

There are alternatives such as using another distribution that satisfies the requirements.

I would generally not recommend using "old" distributions and expect modern container tools to "just work".  Podman as shipped in Ubuntu works and is rock solid and gets updated with each Ubuntu update.  But there is usually a price tag attached if you desire a stable distribution and continuous updates to (container) tools.  That's what enterprise distributions are doing and paid for, in particular for production use cases.

I fully empathize with the desire and the request but I don't find it productive to re-iterate over the same conversation.  I think it's much better to raise the issue of shipping new Podmans to old Ubuntus to the Ubuntu community.  They're doing a great job and maybe Canonical invests resources.

**Comment by StarpTech** [comment]
I totally understand. It's just super sad from the contributor's perspective as well. I use PopOS, and I can't use Podman 4.2 because it's not available on any channel.

> That's what enterprise distributions are doing and paid for, in particular for production use cases.

Didn't know. Which company provides paid distributions for Podman on Ubuntu?

> I fully empathize with the desire and the request but I don't find it productive to re-iterate over the same conversation. I think it's much better to raise the issue of shipping new Podmans to old Ubuntus to the Ubuntu community. They're doing a great job and maybe Canonical invests resources.

Sounds like you have an idea of how to do this. Please explain.

**Comment by lsm5** [comment]
> I totally understand. It's just super sad from the contributor's perspective as well. I use Debian, and I can't use Podman 4.2 because it's not available on any channel.

For Debian 11, I'm checking with the admins of OBS, the place where the repo I mentioned earlier is hosted. There are some dependency issues preventing a successful build currently.

> 
> > That's what enterprise distributions are doing and paid for, in particular for production use cases.
> 
> Didn't know. Which company provides paid distributions for Podman on Ubuntu?

No paid support on Ubuntu, it's entirely volunteer-based support there. Enterprise support is available for Podman on RHEL. There are no other paid alternatives.

> Sounds like you have an idea of how to do this. Please explain.

First step would be to file an issue on ubuntu / debian's bug tracker. The official package maintainer on both distros is @siretart who can provide you with further insight into the debian process. Again, he's generously volunteering his spare time toward this, and combined with debian's own constraints, there will be limits on what can be shipped in the official repositories.

Hope that helps.

**Comment by TCROC** [comment]
I believe I mentioned it earlier and it was said the community could do it if they wanted, but I just thought I'd bring it up again:

Why is using a cross platform packaging software such as Snapcraft as an official distro not being discussed more?  I believe the selling point of these packaging tools is they are supposed to solve a lot of these complexities.  Is there a reason Podman wouldn't want to officially use one?

^ Keep in mind, I'm merely a naive dev who wants to use the shiny toys such as new Podman on my Ubuntu :)

**Comment by TCROC** [comment]
Ah found it.  It got moved to a discussion.  Lives over here now :)

https://github.com/containers/podman/discussions/14360

**Comment by vrothberg** [comment]
> No paid support on Ubuntu, it's entirely volunteer-based support there. Enterprise support is available for Podman on RHEL. There are no other paid alternatives (to my knowledge at least).

Podman is also supported in SUSE's SLES.

Rolling distributions such as Arch or OpenSUSE Tumbleweed naturally update Podman almost instantly.

**Comment by StarpTech** [comment]
Thanks for the many responses! It helps a lot to get a better picture.

**Comment by matt-deboer** [comment]
@lsm5 FYI, I was previously (~1month ago) able to install successfully using the unstable 22.04 repo, but it seems at present to be erroring out installing `podman` with a missing `netavark` dep:
```
The following packages have unmet dependencies:
podman : Depends: netavark (>= 1.0.3-1) but it is not installable
```
Any ideas?

**Comment by lsm5** [comment]
netavark 1.2.0 should now be available on the kubic repo (for x86_64 already, others soon to follow) https://build.opensuse.org/package/show/devel:kubic:libcontainers:unstable/netavark .

cc @cevich

**Comment by cevich** [comment]
Thanks @lsm5 I'll give it a go.

**Comment by matt-deboer** [comment]
Thanks much,  @lsm5 ! successful build/install now on x86_64 22.04

**Comment by alexanderadam** [comment]
Is there any chance also to have the stable versions available somehow or is the integration effort difference between unstable and too huge?

**Comment by lsm5** [comment]
> Is there any chance also to have the stable versions available somehow or is the integration effort difference between unstable and too huge?

@alexanderadam The repo is called (and will continue to be called) `unstable` mainly so people are aware it's not recommended for production use. Only the stable upstream releases are being sent to this `unstable` repo but the packages don't go through debian's or ubuntu's official QE process. I stopped sending RC builds to this repo recently. There might be scope for some automated QE but it's not too high on my priority list atm. I'll gladly take any help from the community toward more testing.

Does that answer your Q or did you mean something else?

**Comment by lsm5** [comment]
> I guess it does. If I understood it correctly, it would be possible to provide a stable channel in general but its priority is low and hence it will currently stay that way that there's production ready solution for Podman versions higher 3.x.

We did provide `stable` and `testing` channels back when I was building deb packages the rather conventional way. Those are not being updated anymore.

RE: production ready, I don't think the podman team will ever be comfy recommending the OBS / kubic packages for production use. Having a `stable` channel would, in my opinion, raise expectations for the deb packages, which won't be in scope for the podman team, at least for the foreseeable future. Even if we have `all tests pass` with the kubic packages, the podman installation docs will still *only* recommend the official distro packages for production use.

So, I guess we are pretty much on the same page. I'm just trying to keep expectations super low :smile:

**Comment by afbjorklund** [comment]
> Have either Red Hat or Canonical shown any signs of stepping up to move Podman to v4 by, say, Ubuntu 24.04?

I think you need to ask the unsung heroes, like @siretart, about the roadmap for the packaging in debian:

https://packages.debian.org/search?keywords=podman&searchon=names&exact=1&suite=all&section=all

Once it is included in debian, that increases the chances of the package being selected for future ubuntu...

https://packages.ubuntu.com/search?keywords=podman&searchon=names&exact=1&suite=all&section=all

**Comment by hkrutzer** [comment]
I think it has been [made quite clear](https://github.com/containers/podman/issues/14302#issuecomment-1191668345) that Red Hat will not support Debian or Ubuntu:

>The Podman maintainers simply don't have the capacity to ship/package new versions to all distributions since the support matrix is just too big.

as well as from the fact that this issue is closed.

One can reach out to Ubuntu as was suggested in the same comment:

>In my view and experience, it's best to work directly together with the distributions. Did you reach out to the Ubuntu community/Canonical and asked whether/how Podman can be updated for LTS versions?

or for example sponsor someone to create the package.

**Comment by mheon** [comment]
@lsm5 is on the Podman team and continues to put a lot of time into packaging Podman for Ubuntu, but only via OBS; all efforts regarding the official repos are handled by @siretart and the Ubuntu maintainers.

**Comment by vrothberg** [comment]
> I think it has been https://github.com/containers/podman/issues/14302#issuecomment-1191668345 that Red Hat will not support Debian or Ubuntu:

I want to clarify just to make sure the text isn't misinterpreted without reading the linked comment.  I also want to avoid that we generalize too much.

"Support" can mean many things.  As an _upstream_ project, we support all distributions equally.  Bugs will be fixed and features will be implemented regardless of the distribution.

What this issue is about is _package maintenance_. There is an overlap of upstream contributors and downstream packagers but it is practically impossible for upstream to maintain packages for all distributions.  Some users want it stable, some users want it rolling, etc. and all have different life and release cycles.

Many upstream contributors are Red Hat employees, so Fedora is really quick in consuming new versions. But ArchLinux,  OpenSUSE and others are just as fast.

So what people in this thread are asking for is accelerating downstream packaging for Debian and Ubuntu. Doing it faster and providing stable and fast/rolling versions of the packages is a lot of work that will exceed volunteers' and upstream capabilities.

As mentioned above, I recommend reaching out to the Debian/Ubuntu community and help them. Or open a ticket at Ubuntu/Canonical. Really, please talk to them so they realize a _business need_ to provide Podman support for their products.

**Comment by afbjorklund** [comment]
And it would probably still be a good idea to update to podman 3.4.7, even if it is not supported anymore either.

**Comment by lsm5** [comment]
@jeremyn sure, that's best done on https://github.com/containers/podman.io

**Comment by TCROC** [comment]
@jeremyn YES!  This got me the up to date podman!  Should def be the recommended way for Ubuntu.  Here are the steps:

```bash
sudo apt install build-essentials
sudo apt install uidmap
```

Install homebrew: https://brew.sh/

```bash
brew install podman
```

ez pz! :)

**Comment by dawidd6** [comment]
Good to know it's working for others too :)

**Comment by siretart** [comment]
Thank you for all the support and kind words on distro packaging.

Maybe a quick status update, I've uploaded podman 4.3.1 to debian/experimental yesterday. I'd really appreciate if people could give those packages a shot and file bugs using `reportbug`, the debian bug submission tool. Bonus points if you could help me with the open bugs at https://bugs.debian.org/src:libpod  -- just sent a followup at NNNN@bugs.debian.org (with NNNN being the bug number) whether you can reproduce it or not.

I plan on uploading 4.3.1 to unstable later this week. This will give the packages much more attention, and I expect more bugs to surface.

As for ubuntu, the packages are mostly in sync with debian/unstable, but ubuntu packages runc and docker differently, which causes an unfortunate delta that needs to be handled with manually. Nothing unsurmountable, just annoying. 

I keep getting asked about backports to bullseye (and by extension, to released versions of ubuntu). The short answer is that I don't have capacity for this. It is not "just" about podman, it is the whole container stack: containers/storage, containers/image, containers/common, containers/buildah and containers/podman, all of which need to be backported in that order. Also, containers/storage has a number of system-related dependencies that may or may not need to be backported as well. It is a less than trivial backport.

**Comment by TCROC** [comment]
> @jeremyn YES! This got me the up to date podman! Should def be the recommended way for Ubuntu. Here are the steps:
> 
> ```shell
> sudo apt install build-essentials
> sudo apt install uidmap
> ```
> 
> Install homebrew: https://brew.sh/
> 
> ```shell
> brew install podman
> ```
> 
> ez pz! :)

Update regarding using homebrew.  I found that homebrew actually installed a bunch of dependencies that were already installed on my Ubuntu distro causing conflicts.  It would be nice if it detected software already installed on my pc.  Would this be more appropriate as a podman or homebrew issue?

**Comment by rhatdan** [comment]
Yes this has nothing to do with the upstream podman effort.

**Comment by uogbuji** [comment]
This is a complete dead end for me. Trying to use VS Code & Running into 

```
Error: unable to connect to Podman socket: server API version is too old. Client "4.0.0" server "3.4.4"
```

No I'm not installing homebrew on my Linux Mint box, just to get a year-old version of Podman. I guess it's back to Docker.

**Comment by hkrutzer** [comment]
<img width="564" alt="image" src="https://github.com/containers/podman/assets/3464899/68207ac9-a60a-4c4e-bdf7-ea8318b6bb7e">

https://launchpad.net/ubuntu/+source/libpod

**Comment by uogbuji** [comment]
So I guess until Linux Mint is updated to Lunar Lobster, or the Kubic PPA is supported again, I'd have to either build podman from scratch or use homebrew.  *sigh* Look, I'm not meaning to be difficult, but I'm just trying to get work done and have already burned a couple of hours on this. I don't really have a choice but to just use Docker, which I already know will work with VS Code remote with no fiddling. Respect to you, though, @hkrutzer, for replying to my peeved note.

**Comment by lsm5** [comment]
> So I guess until Linux Mint is updated to Lunar Lobster, or the Kubic PPA is supported again, I'd have to either build podman from scratch or use homebrew. _sigh_ Look, I'm not meaning to be difficult, but I'm just trying to get work done and have already burned a couple of hours on this. I don't really have a choice but to just use Docker, which I already know will work with VS Code remote with no fiddling. Respect to you, though, @hkrutzer, for replying to my peeved note.

the kubic `unstable` repo does have the latest v4.5.1 https://build.opensuse.org/package/show/devel:kubic:libcontainers:unstable/podman .  But beware, use in production at your own risk. That's the extent of kubic support the podman team will be able to provide for the foreseeable future.
